x-pack/plugins/lens/public/visualizations/xy/visualization.tsx

```diff
@@ -788,72 +798,43 @@ export const getXyVisualization = ({
         }
 
         const errorMessages = getAnnotationLayerErrors(layer, annotation.id, dataViews);
-        errors.push(
-          ...errorMessages.map((errorMessage) => {
-            const message: UserMessage = {
-              severity: 'error',
-              fixableInEditor: true,
-              displayLocations: [
-                { id: 'visualization' },
-                { id: 'dimensionButton', dimensionId: annotation.id },
-              ],
-              shortMessage: errorMessage,
-              longMessage: (
-                <FormattedMessage
-                  id="xpack.lens.xyChart.annotationError"
-                  defaultMessage="Annotation {annotationName} has an error: {errorMessage}"
-                  values={{
-                    annotationName: annotation.label,
-                    errorMessage,
-                  }}
-                />
-              ),
-            };
-            return message;
-          })
-        );
+        errors.push(...errorMessages);
       });
     });
 
-    // Data error handling below here
-    const hasNoAccessors = ({ accessors }: XYDataLayerConfig) =>
-      accessors == null || accessors.length === 0;
-
-    const hasNoSplitAccessor = ({ splitAccessor, seriesType }: XYDataLayerConfig) =>
-      seriesType.includes('percentage') && splitAccessor == null;
-
     // check if the layers in the state are compatible with this type of chart
     if (state && state.layers.length > 1) {
       // Order is important here: Y Axis is fundamental to exist to make it valid
-      const checks: Array<[string, (layer: XYDataLayerConfig) => boolean]> = [
-        ['Y', hasNoAccessors],
-        ['Break down', hasNoSplitAccessor],
-      ];
-
-      for (const [dimension, criteria] of checks) {
-        const result = validateLayersForDimension(dimension, state.layers, criteria);
-        if (!result.valid) {
-          errors.push({
-            severity: 'error',
-            fixableInEditor: true,
-            displayLocations: [{ id: 'visualization' }],
-            shortMessage: result.payload.shortMessage,
-            longMessage: result.payload.longMessage,
-          });
-        }
+      const yLayerValidation = validateLayersForDimension(
```

Comment:
Not sure the reason of this full rewrite, which could have been just a:

```ts
errors.push(result.error);
```

Was it there a good reason to revisit it?

Reply:
Readability mainly. We created an array of type + accessors to iterate over it and call the validate function.
I considered this a bit too much for something that doesn't change: directly calling the validation and moving the accessor definition in the arguments reduce the number of variable names we need to create (`checks`, `hasNoAccessor` etc) and also cleans up a bit the typing for `validateLayersForDimension`